twine/commands/upload.py

```diff
@@ -139,6 +139,9 @@ def upload(upload_settings: settings.Settings, dists: List[str]) -> None:
             continue
 
         resp = repository.upload(package)
+        logger.info(f"Response from {resp.url}:\n{resp.status_code} {resp.reason}")
+        if resp.text:
+            logger.info(resp.text)
```

Comment:
After dithering on it for awhile, I went back to a simple log message, that honestly feels like it will be redundant in most cases, but potentially useful. Is there other information that should be included at this level of verbosity?

Failure:

```
Uploading example_pkg_bhrutledge-0.0.5-py3-none-any.whl
Response from https://test.pypi.org/legacy/:
400 File already exists. See https://test.pypi.org/help/#file-name-reuse for more information.
<html>
 <head>
  <title>400 File already exists. See https://test.pypi.org/help/#file-name-reuse for more information.</title>
 </head>
 <body>
  <h1>400 File already exists. See https://test.pypi.org/help/#file-name-reuse for more information.</h1>
  The server could not comply with the request since it is either malformed or otherwise incorrect.<br/><br/>
File already exists. See https://test.pypi.org/help/#file-name-reuse for more information.


 </body>
</html>
HTTPError: 400 Bad Request from https://test.pypi.org/legacy/
File already exists. See https://test.pypi.org/help/#file-name-reuse for more information.
```

Success:

```
Uploading example_pkg_bhrutledge-0.0.6.post2-py3-none-any.whl
Response from https://test.pypi.org/legacy/:
200 OK
Uploading example-pkg-bhrutledge-0.0.6.post2.tar.gz
Response from https://test.pypi.org/legacy/:
200 OK

Reply:
I do still wonder if we need any information (for non-PyPI artifact hosts) regarding redirects. Even if it's a count and not the specific status codes in the history.

Reply:
@sigmavirus24 I wondered about that, but I thought maybe it was moot because redirects result in an error:

https://github.com/pypa/twine/blob/a228795a81fce3e98421c82ff844c89d876ff94c/twine/commands/upload.py#L146-L154

But, I could certainly be wrong. If so, can you recommend some specific attributes of the `Response` object to log?

Reply:
Ah, nevermind. That makes sense to me that we've turned off redirects.

Reply:
If we did allow redirects, we could log `len(response.history)` to get the number of redirects. Alternatively, we could do a `', '.join(str(r.status_code) for r in response.history)`, etc.